### PR TITLE
Enable source debugging feature on windows

### PR DESCRIPTION
**Pull request by xujuntwt95329**
- use platform independent data types in debug-engine library
- add os_socket APIs and provide windows and posix implementation
- avoid using platform related header files in non-platform layer
- use format specifiers macros for sprintf and sscanf

### CODE REVIEW

File: core/iwasm/libraries/debug-engine/handler.c [hunk 1]
```diff
@@ -295,40 +296,40 @@ send_thread_stop_status(WASMGDBServer *server, uint32_t status, uint64_t tid)
...
+                   "nobody");
     if (tids_number > 0) {
         len += sprintf(tmpbuf + len, "threads:");
         while (i < tids_number) {
             if (i == tids_number - 1)
-                len += sprintf(tmpbuf + len, "%lx;", tids[i]);
+                len += sprintf(tmpbuf + len, "%" PRIx64 ";", tids[i]);
             else
-                len += sprintf(tmpbuf + len, "%lx,", tids[i]);
+                len += sprintf(tmpbuf + len, "%" PRIx64 ",", tids[i]);
```
Comment (wenyongh) [inline core/iwasm/libraries/debug-engine/handler.c:315]: snprintf?

File: core/iwasm/libraries/debug-engine/handler.c [hunk 2]
```diff
@@ -295,40 +296,40 @@ send_thread_stop_status(WASMGDBServer *server, uint32_t status, uint64_t tid)
...
     else if (status == WAMR_SIG_SINGSTEP) {
-        len += sprintf(tmpbuf + len, "thread-pcs:%lx;00:%s,reason:%s;", pc,
-                       pc_string, "trace");
+        len += sprintf(tmpbuf + len, "thread-pcs:%" PRIx64 ";00:%s,reason:%s;",
+                       pc, pc_string, "trace");
     }
     else if (status > 0) {
-        len += sprintf(tmpbuf + len, "thread-pcs:%lx;00:%s,reason:%s;", pc,
-                       pc_string, "signal");
+        len += sprintf(tmpbuf + len, "thread-pcs:%" PRIx64 ";00:%s,reason:%s;",
```
Comment (wenyongh) [inline core/iwasm/libraries/debug-engine/handler.c:331]: snprintf?

File: core/iwasm/libraries/debug-engine/handler.c [hunk 3]
```diff
@@ -607,7 +614,7 @@ handle_malloc(WASMGDBServer *server, char *payload)
             (WASMDebugInstance *)server->thread->debug_instance, size,
             map_port);
         if (addr) {
-            sprintf(tmpbuf, "%lx", addr);
+            sprintf(tmpbuf, "%" PRIx64, addr);
```
Comment (wenyongh) [inline core/iwasm/libraries/debug-engine/handler.c:617]: snprintf?

File: core/iwasm/libraries/debug-engine/debug_engine.c [hunk 4]
```diff
@@ -512,9 +525,9 @@ wasm_debug_instance_get_thread_status(WASMDebugInstance *instance, uint64 tid)
 }
 
 void
-wasm_debug_instance_set_cur_thread(WASMDebugInstance *instance, uint64 tid)
+wasm_debug_instance_set_cur_thread(WASMDebugInstance *instance, korp_tid tid)
 {
-    instance->current_tid = tid;
+    instance->current_tid = (korp_tid)(korp_tid)tid;
```
Comment (wenyongh) [inline core/iwasm/libraries/debug-engine/debug_engine.c:530]: No need to add `(korp_tid)(korp_tid)`

File: core/iwasm/libraries/debug-engine/gdbserver.c [hunk 5]
```diff
@@ -180,13 +134,13 @@ handler_packet(WASMGDBServer *server, char request, char *payload)
...
 process_packet(WASMGDBServer *server)
 {
-    uint8_t *inbuf = server->pkt.buf;
-    int inbuf_size = server->pkt.size;
-    uint8_t *packetend_ptr = (uint8_t *)memchr(inbuf, '#', inbuf_size);
-    int packetend = packetend_ptr - inbuf;
+    uint8 *inbuf = server->pkt.buf;
+    int32 inbuf_size = server->pkt.size;
+    uint8 *packetend_ptr = (uint8 *)memchr(inbuf, '#', inbuf_size);
+    int32 packetend = (int32)(uintptr_t)(packetend_ptr - inbuf);
```
Comment (wenyongh) [inline core/iwasm/libraries/debug-engine/gdbserver.c:140]: How about rename packetend to packet_size?

File: core/iwasm/libraries/debug-engine/handler.c [hunk 6]
```diff
@@ -589,7 +663,7 @@ handle_malloc(WASMGDBServer *server, char *payload)
         return;
```
Comment (wenyongh) [inline core/iwasm/libraries/debug-engine/handler.c:663]: Should os_mutex_unlock before return

File: core/iwasm/libraries/debug-engine/debug_engine.c (hunk 4 above)
Comment (xujuntwt95329) [reply core/iwasm/libraries/debug-engine/debug_engine.c]: Done

File: core/iwasm/libraries/debug-engine/gdbserver.c (hunk 5 above)
Comment (xujuntwt95329) [reply core/iwasm/libraries/debug-engine/gdbserver.c]: Done

File: core/iwasm/libraries/debug-engine/handler.c (hunk 6 above)
Comment (xujuntwt95329) [reply core/iwasm/libraries/debug-engine/handler.c:660]: Done, thanks!

**Pull request merged**